oscar/slack-bot/bedrock.py

```diff
@@ -0,0 +1,176 @@
+        """
+        Initialize Bedrock knowledge base.
+        
+        Args:
+            region: AWS region for Bedrock service, defaults to config value if None
+        """
+        self.region = region or config.region
+        self.client = boto3.client('bedrock-agent-runtime', region_name=self.region)
+        self.knowledge_base_id = config.knowledge_base_id
+        self.model_arn = config.model_arn
+        self.prompt_template = config.prompt_template
+    
+    def _create_request(self, query: str, session_id: Optional[str] = None) -> Dict[str, Any]:
+        """
+        Create a request for the Bedrock knowledge base.
+        
+        Args:
+            query: The user's query
+            session_id: Optional session ID for maintaining conversation context
+            
+        Returns:
+            A dictionary containing the request parameters
+        """
+        # Check if we're using an inference profile ARN
+        is_inference_profile = "inference-profile" in self.model_arn
+        
+        # Prepare base request structure
+        request = {
+            'input': {'text': query},
+            'retrieveAndGenerateConfiguration': {
+                'type': 'KNOWLEDGE_BASE',
+                'knowledgeBaseConfiguration': {
+                    'knowledgeBaseId': self.knowledge_base_id,
+                    'modelArn': self.model_arn,
+                    'generationConfiguration': {
+                        'promptTemplate': {
+                            'textPromptTemplate': self.prompt_template
+                        }
+                    }
+                }
+            }
+        }
+        
+        # Add query decomposition configuration for non-inference profiles
+        if not is_inference_profile:
+            request['retrieveAndGenerateConfiguration']['knowledgeBaseConfiguration']['orchestrationConfiguration'] = {
+                'queryTransformationConfiguration': {
+                    'type': 'QUERY_DECOMPOSITION'
+                }
+            }
+        
+        # Add session ID if available
+        if session_id:
+            request['sessionId'] = session_id
+        
+        return request
+    
+    def query(self, query: str, session_id: Optional[str] = None, 
+              context_summary: Optional[str] = None) -> Tuple[str, Optional[str]]:
+        """
+        Query Bedrock knowledge base with session or context.
+        
+        Args:
+            query: The user's query to the knowledge base
+            session_id: Optional session ID for maintaining conversation context
+            context_summary: Optional summary of previous conversation context
+            
+        Returns:
+            A tuple containing (response_text, session_id)
+            
+        Raises:
+            Exception: If both the primary query and fallback query fail
+        """
+        # Build prompt with context if available
+        if context_summary and not session_id:
```

Comment:
Could you please verify the case when context_summary and session_id both exist in dynamo, but session_id already expired in bedrock. Will this cause bedrock to error out, or will it provide you with a new session id to update?

Thanks.